beater/middleware/rate_limit_middleware.go

```diff
@@ -0,0 +1,38 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package middleware
+
+import (
+	"github.com/elastic/apm-server/beater/api/ratelimit"
+	"github.com/elastic/apm-server/beater/config"
+	"github.com/elastic/apm-server/beater/request"
+)
+
+const burstMultiplier = 3
```

Comment:
If making this generic then the `burstMultiplier` should also be a param instead of a const.

Reply:
why?

Reply:
Because it makes the cache creation more flexible and we might not want to use the same burst multiplier for different rate limiters.

Reply:
So are you suggesting to make a config option for that? I'd rather don't overwhelm users with more settings until proof they are actually needed, and tackle that problem when we have it. This is for now an implementation detail.

If anything, it _could_ be relevant a different rate limit per this endpoint, but that is out of scope here.

Reply:
Definitly not suggesting a config option. If you feel so strongly against making this a parameter to the middleware, leave it as is.